### ISSUE TITLE
Enqueue jobs to GCP transparently by leveraging ActiveJob's abstractions, and execute these

### ISSUE THREAD

**Issue by ce07c3** [artygus/scheduler#1]
When a job is scheduled to perform at a later time, we can ask a background job processor to do the job. With Rails 5 (?), we now have an abstraction on top of Rails called ActiveJob. Use ActiveJob to enqueue jobs, and dequeue these using a job processor of your choice built on top of Google Cloud Pub/Sub (emulator is probably easier to use).

- [x] Enqueue using ActiveJob (adapter)
- [x] Send to GCP Pub/Sub
- [x] Dequeue using background job processor
- [x] Execute using a thread pool or whatever feels best

We can augment the implementation by collecting some metrics:

- Execution duration
- Number of executed jobs

using ActiveSupport::Instrumentation.

Let's work together here and on Skype. :)

**Comment by ce07c3** [comment]
Adapter needs to be built for ActiveJob - there should be a few default adapters built into Rails already @artygus?

**Comment by artygus** [comment]
what about Active Job Async adapter?

**Comment by ce07c3** [comment]
Yeah, that's one of them. We need to hook this up with our GCP setup. Any ideas?

**Comment by artygus** [comment]
google-cloud-ruby would do to handle interaction with gcp

**Comment by ce07c3** [comment]
👍 Yes. How would you enqueue the jobs to GCP Pub/Sub? Let's write some code, if you can open a PR for this?

**Comment by artygus** [comment]
sure, I'll handle this

**Comment by ce07c3** [comment]
I looked into it a bit - we need to serialize the job in some way. Should we use JSON @artygus?

**Comment by artygus** [comment]
yeah, I think so

**Comment by ce07c3** [comment]
Ok. It seems we can use `ActiveJob::Job`'s `serialize` to get a hash of its values, we just need to make sure we can turn it into a job again once we dequeue it.

**Comment by artygus** [comment]
take a look at #1, it's a basic class can queue and pop messages

**Comment by ce07c3** [comment]
Added a comment there.
